data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,197 @@
+    return data in ["-", "+", "*", "^", "/"]
 
 
-def solve(post_fix):
+def evaluate(post_fix: list, verbose: bool = False) -> int | float | str | None:
+    """
+    Function that evaluates postfix expression using a stack.
+    >>> evaluate(["2", "1", "+", "3", "*"])
+    9
+    >>> evaluate(["4", "13", "5", "/", "+"])
+    6.6
+    >>> evaluate(["2", "-", "3", "+"])
+    1
+    >>> evaluate([])
+    0
+
+    Parameters
+    ----------
+    post_fix : list
+        The postfix expression tokenized into operators and operands and stored as a
+        python list
+
+    verbose : bool
+        Display stack contents while evaluating the expression if verbose is True
+
+    Returns
+    -------
+    int
+        The evaluated value
+    """
     stack = []
-    div = lambda x, y: int(x / y)  # noqa: E731 integer division operation
     opr = {
-        "^": op.pow,
-        "*": op.mul,
-        "/": div,
-        "+": op.add,
-        "-": op.sub,
+        "^": lambda p, q: p**q,
+        "*": lambda p, q: p * q,
+        "/": lambda p, q: p / q,
+        "+": lambda p, q: p + q,
+        "-": lambda p, q: p - q,
     }  # operators & their respective operation
+    if len(post_fix) > 0:
+        if verbose:
+            # print table header
+            print("Symbol".center(8), "Action".center(12), "Stack", sep=" | ")
+            print("-" * (30 + len(post_fix)))
+        for x in post_fix:
+            is_number, x = get_number(x)
+            if is_number:  # if x is a number (integer, float)
+                stack.append(x)  # append x to stack
+                if verbose:
+                    # output in tabular format
+                    print(
+                        str(x).rjust(8),
+                        ("push(" + str(x) + ")").ljust(12),
+                        stack,
+                        sep=" | ",
+                    )
+            elif is_operator(x):
+                # If only 1 value is inside stack and + or - is encountered
+                # then this is unary + or - case
+                if x in ["-", "+"] and len(stack) < 2:
+                    b = stack.pop()  # pop stack
+                    if x == "-":
+                        stack.append(-b)  # negate b and push again into stack
+                    else:  # when x is unary +
+                        stack.append(b)
+                    if verbose:
+                        # output in tabular format
+                        print(
+                            "".rjust(8),
+                            ("pop(" + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+                        print(
+                            str(x).rjust(8),
+                            ("push(" + str(x) + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+                else:
+                    b = stack.pop()  # pop stack
```

Comment:
You can use continue statements to avoid nesting code inside else blocks:
```suggestion
                        )
                    continue
                
                # x is a binary operator
                b = stack.pop()  # pop stack
```